dace/sdfg/validation.py

```diff
@@ -28,6 +28,9 @@ def validate_sdfg(sdfg: 'dace.sdfg.SDFG'):
         on failure.
     """
     try:
+        if sdfg.parent is not None and sdfg.parent_nsdfg_node not in sdfg.parent.nodes():
+            raise InvalidSDFGError("Invalid parent or parent_nsdfg_node", sdfg, None)
+
```

Comment:
Since we have the parent and the node, we could just pass them along to `validate_sdfg` (kwargs with None defaults) to avoid looping over the parent's nodes.